services/requester/cadence/call.cdc

```diff
@@ -1,20 +1,13 @@
 import EVM
 
 access(all)
-fun main(hexEncodedData: String, hexEncodedAddress: String): String {
+fun main(hexEncodedTx: String): String {
     let account = getAuthAccount<auth(Storage) &Account>(Address(0xCOA))
 
     let coa = account.storage.borrow<auth(EVM.Call) &EVM.CadenceOwnedAccount>(
         from: /storage/evm
     ) ?? panic("Could not borrow reference to the COA!")
-    let addressBytes = hexEncodedAddress.decodeHex().toConstantSized<[UInt8; 20]>()!
+    let txResult = EVM.run(tx: hexEncodedTx.decodeHex(), coinbase: coa.address())
 
-    let callResult = coa.call(
-        to: EVM.EVMAddress(bytes: addressBytes),
-        data: hexEncodedData.decodeHex(),
-        gasLimit: 15000000, // todo make it configurable, max for now
-        value: EVM.Balance(attoflow: 0)
-    )
-
-    return String.encodeHex(callResult.data)
+    return String.encodeHex(txResult.data)
```

Comment:
We can potentially merge the `call.cdc` and `estimate_gas.cdc` scripts, if we return the entire `txResult` struct. To do that though, we should add a decoder for the `EVM.Result` struct.

Reply:
yes, that's a future todo, maybe add it as an issue or in comment